app/assets/javascripts/school_administrator_dashboard/dashboard_bar_chart.jsx

```diff
@@ -47,18 +50,35 @@ export default React.createClass({
     };
   },
 
+  shouldComponentUpdate: function(nextProps) {
```

Comment:
@edavidsonsawyer Can you say a little more about what this is doing? How is this different than the normal `shouldComponentUpdate` behavior? I tried removing it in my local machine and the dashboard chart seemed to work the same as before, unless I'm missing a detail that's changed.

Reply:
Right, I meant to add a comment here. This is to prevent all the charts from redrawing when a user clicks on the column to get the list of students by homeroom. The list is the only thing that needs to change so it shouldn't trigger a render of the charts.

Reply:
Ooooh that's really interesting. The way the school overview page works, if you click one filter, **everything** on the page responds to that change. The percentages for each of the slice table categories, the mini bar charts, and the students table — they all update. I wonder if users will expect other pages like the schoolwide attendance dashboard to perform the same way? In that case, would it make sense to redraw the monthly charts when a new homeroom is selected?

Reply:
That's a good point, @alexsoble. Here's why I'm inclined to think these are different cases. In the school overview the component, even though everything redraws, that feels natural because every part of the page has the potential to be affected by the filter you select. It's almost reassuring to see the app is recalculating each point to be sure it's correct with the new filter. 

Here, though, the user isn't doing anything to the data displayed by the chart. They just want more information about one of the data points it displays. If the chart redraws at that point, they might assume they accidentally changed the chart parameters. When I see a chart redraw I assume that's because the app thought the chart might need to change because of something I did. I found the redrawing to be jarring even though I know what's happening. 

So that was my rationale. Thoughts?

Reply:
Hm. Now that I think about this a little more I think I might want to add an intermediate component since it's actually doing the work to come up with the chart data even though the chart doesn't use it. So even if we want to keep the conditional update in, hold off on this PR so I can make sure we aren't bothering to do more work than we need to.

Reply:
Actually, scratch that. The changes didn't affect performance as much as I'd hoped so I'll see if I can handle improving the student chart speed separately.

Reply:
Interesting! User testing will definitely help us make decisions here. As far as the code, I think either let's add a code comment to this lifecycle function explaining why it's here, or implement an intermediate component. First one would let us ship faster but long run I think this `shouldComponentUpdate` override is telling us there should be another component here.

Reply:
Alright, @alexsoble. For now I'll add the comment since I think I'd have to overwrite the method even with the other changes I was considering. I'll be interested to see how users respond. It's easy enough to change if redrawing the charts makes for a better experience!